scripts/generate-sitemap.js

```diff
@@ -14,28 +17,90 @@ const siteMetadata = require('../data/siteMetadata')
+          .replace(`.${loc}`, '')
+          .replace('/feed', '')
+          .replace('.xml', '')),
+      loc,
+      false, // Indicate if the element is already present or not
+    ])
+
   const sitemap = `
-        <?xml version="1.0" encoding="UTF-8"?>
-        <urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
-            ${pages
-              .map((page) => {
-                const path = page
-                  .replace('pages/', '/')
-                  .replace('data/blog', '/blog')
-                  .replace('public/', '/')
-                  .replace('.js', '')
-                  .replace('.mdx', '')
-                  .replace('.md', '')
-                  .replace('/feed.xml', '')
-                const route = path === '/index' ? '' : path
-                if (page === `pages/404.js` || page === `pages/blog/[...slug].js`) {
+  <?xml version="1.0" encoding="UTF-8"?>
+  <urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9"
+      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+      xmlns:xhtml="http://www.w3.org/1999/xhtml"
+      xsi:schemaLocation="http://www.sitemaps.org/schemas/sitemap/0.9
+      http://www.sitemaps.org/schemas/sitemap/0.9/sitemap.xsd
+      http://www.w3.org/1999/xhtml
+      http://www.w3.org/2002/08/xhtml/xhtml1-strict.xsd">
+            ${pagesWithLoc
+              .map(([path, loc, alreadyPresent]) => {
+                // @todo: Can you check especially here ?
+                const route = path.includes('/index') ? path.replace('/index', '') : path
+                if (
+                  path.includes(`/404.js`) ||
+                  path.includes(`/blog/[...slug].js`) ||
+                  alreadyPresent
+                ) {
+                  // Not sure about the [...slug] condition...
                   return
                 }
-                return `
+                const routeMultiLang = pagesWithLoc.filter(
+                  ([ipath, iloc, _]) => ipath.replace(`/${iloc}`, '') == path.replace(`/${loc}`, '')
+                )
+                routeMultiLang.map((e) => (e[2] = true)) //making allreadyPresnt to true
+                if (routeMultiLang.length === 1)
+                  return `
                         <url>
                             <loc>${siteMetadata.siteUrl}${route}</loc>
                         </url>
                     `
+                return `
+                          <url>
+                              <loc>${siteMetadata.siteUrl}${
+                  routeMultiLang.filter(([path, loc]) => (loc === defaultLocale ? path : ''))[0][0]
+                }</loc>
+                  ${routeMultiLang.map(
+                    ([xe, xloc]) =>
+                      `
+                               <xhtml:link 
+                               rel="alternate"
+                               hreflang="${xloc}"
+                               href="${siteMetadata.siteUrl}${xe}"/>
+                               `
+                  )}
```

Comment:
Perhaps change this line to`).join('')}`? The comma looks a little off in the XML file though it does not throw any error in the validators I tried.